pygfx/utils/text/_fontfinder.py

```diff
@@ -9,37 +9,98 @@
 import time
 import secrets
 
+import freetype
+
 from .. import logger, get_resources_dir, get_cache_dir
 
 
-def find_fonts_from_dir(directory, recursive):
-    """
-    Return a list of all fonts matching any of the extensions, found
-    recursively under the directory.
-    """
-    if not os.path.isdir(directory):
-        raise OSError(f"Not a directory: {directory}")
-    extensions = ".ttf", ".otf"
-    dir_paths = set()
-    file_paths = set()
-    if not recursive:
-        dir_paths.add(directory)
-        for fname in os.listdir(directory):
-            if fname.lower().endswith(extensions):
-                file_paths.add(os.path.join(directory, fname))
-    else:
-        for dirpath, _, filenames in os.walk(directory):
-            dir_paths.add(dirpath)
-            for fname in filenames:
-                if fname.lower().endswith(extensions):
-                    file_paths.add(os.path.join(dirpath, fname))
-    return dir_paths, file_paths
+class FontFile:
+    """Object to represent a font file."""
+
+    def __init__(self, filename, family_name=None, style_name=None, codepoints=None):
+        assert isinstance(filename, str)
+        assert family_name is None or isinstance(family_name, str)
+        assert style_name is None or isinstance(style_name, str)
+        assert codepoints is None or isinstance(codepoints, set)
+
+        self._filename = filename
+        self._family_name = family_name
+        self._style_name = style_name
+        self._name = None
+        self._codepoints = codepoints
+
+    def __repr__(self):
+        return f"<FontFile {self.name} at 0x{hex(id(self))}>"
+
+    def __hash__(self):
+        return hash(self.name)
```

Comment:
What if two different versions of the same font are found?

Reply:
Then one will be used. This'd be either the first or the last added. Would be good to make this behavior explicit and document it.

Reply:
This will be `logger.debug()`'ed.